joomla/joomla-cms
# Check for component container item in admin menu recovery mode

### Summary of Changes
First, this PR reverts #13838 because it's inconsistent from a user perspective: the new admin menu has a built-in recovery mode that checks the presence of important menu items and offers a link that temporarily restores the old menu structure. With #13838 an additional check for the component container item was added - but not to the recovery mode, but to the general code of the module causing the item to show up without any explanation.

This PR moves the check for the container to the general recovery-mode logic and adjusts the messages accordingly, giving a consistent workflow.

### Testing Instructions
* Create a custom admin menu
* Set it as your active menu in menu module settings.
* A message stating that the recommended menu items for Modules, Menus and Components are missing should be shown including a link to enable the recovery mode.
* Create a new menu item of type "System Links > Components Menu Container" in your custom menu.
* The message regarding the missing component item should be gone.

### Documentation Changes Required
None
